lua/wire/wirenet.lua

```diff
@@ -207,4 +207,226 @@ function Net.Trivial.Receive(name, callback)
+								numEntries = numEntries + 1
+							else
+								-- Delete the entIndex
+								ss.index = ss.index - 1
+								ss[ss.index] = nil
+							end
+
+							if sizeleft <= 0 then self:sync() break end
+						end
+					else
+						self.entities[ent] = nil
+					end
+				end
+				return numEntries
+			end) > 0 then
+				self:send(ss:getString())
+			end
+
+		end,
+		fullsnapshot = function(self, ply)
+			local snapshots = {}
+			local curent = nil
+
+			while true do
+				local shouldBreak = true
+				local ss = StringStream()
+				local sizeleft = net.Stream.SendSize*net.Stream.MaxServerChunks*0.5
+
+				if writeUnboundedArray(ss, function()
+					local numEntries = 0
+					for ent, mem in next, self.entities, curent do
+						if ent:IsValid() then
+
+							ss:writeInt16(ent:EntIndex())
+							local written = writeUnboundedArray(ss, function()
+								local startpos = ss.index
+								ent:SerializeFullMemory(ss, sizeleft)
+								return calcWritten(ss, startpos)
+							end)
+
+							if written>0 then
+								sizeleft = sizeleft - written - 6
+								numEntries = numEntries + 1
+								if sizeleft <= 0 then
+									shouldBreak = false
+									curent = ent
+									break
+								end
+							else
+								-- Delete the entIndex
+								ss.index = ss.index - 1
+								ss[ss.index] = nil
+							end
+
+						else
+							self.entities[ent] = nil
+						end
+					end
+					return numEntries
+				end) > 0 then
+					snapshots[#snapshots + 1] = ss:getString()
+				end
+				if shouldBreak then break end
+			end
+
+			for _, v in ipairs(snapshots) do
+				self:send(v, ply, #snapshots)
+			end
+		end,
+		send = function(self, data, ply, snapshots)
+			net.Start("WirelibSyncEntities")
+			net.WriteUInt(self.syncid, 16)
+			if ply then
+				net.WriteBool(true)
+				net.WriteUInt(snapshots, 32)
+				net.WriteStream(data)
+				net.Send(ply)
+			else
+				net.WriteBool(false)
+				local newid = (self.syncid + 1) % 65536
+				self.syncid = newid
+				net.WriteStream(data, function() if newid==self.syncid then self.sending = false end end)
+				net.Broadcast()
+				self.sending = true
+			end
+		end,
+		receive = function(self)
+			local snapshots
+			local id = net.ReadUInt(16)
+			if net.ReadBool() then
+				snapshots = net.ReadUInt(32)
+			end
+			net.ReadStream(function(data)
+				self:applyReceivedData(id, data, snapshots)
+			end)
+		end,
+		applyReceivedData = function(self, id, data, snapshots)
+			if id==self.syncid or snapshots then
+				if snapshots then
+					self.snapshotsreceived = self.snapshotsreceived + 1
+					if self.snapshotsreceived == snapshots then
+						self.syncid = id
+					end
+				else
+					self.syncid = (self.syncid + 1) % 65536
+				end
+				if data then
+					local ss = StringStream(data)
+					for i=1, ss:readUInt32() do
+						local ent = Entity(ss:readUInt16())
+						local size = ss:readUInt32()
+						if ent:IsValid() and ent.DeserializeMemory then
+							ent:DeserializeMemory(ss)
```

Comment:
Should seek based on size just in case DeserializeMemory reads too much by mistake.